src/PollController.js

```diff
@@ -42,23 +43,38 @@ function (Okta, FormController) {
       modelEvents: {
         'cancelRequest': '_stopPolling'
       },
-
+      formChildren: [
+        FormType.View({
+          View:
+             '<div >\
+               <div data-se="poll-waiting" class="hide okta-waiting-spinner"></div>\
+             </div>'
+        })
+      ],
       _startPolling: function () {
         // start polling request 
-        this.transaction.poll()
+        this.transactionObject.poll()
           .then((resp) => {
-            this.options.appState.set('transaction', resp);
-            const factorPollingInterval = resp.factor.profile.refresh;
+            this.options.appState.trigger('setTransaction', resp);
```

Comment:
the `transaction` is just a field that is being sent in the response.
I am actually going to make a small update here. One thing @aarongranick-okta pointed out to me was that we probably don't need to set this new transaction if our status has not changed, to prevent a re-render.

We need to set the transaction so that when this changes to success we can proceed and render a different controller accordingly.

TLDR: I will update this to 

```  
 if (resp.status === 'SUCCESS') {
              this.$('.okta-waiting-spinner').hide();
              this.options.appState.set('transaction', resp);
            } 
```